include/openspace/engine/openspaceengine.h

```diff
@@ -65,6 +65,15 @@ struct CommandlineArguments {
 
 class OpenSpaceEngine {
 public:
+    // A mode that specifies which part of the system is currently in control.
+    // The mode can be used to limit certain features, like setting time, navigation
+    // or triggering scripts
+    enum class Mode {
```

Comment:
I like these different modes, but I think it should be named `InteractionMode` or something. From elsewhere in the code, the mode of the OpenSpaceEngine would seem broad from a readability standpoint.

Reply:
I sort of agree, and like the name `ÌnteractionMode`. Good suggestion

Discussed this with @alexanderbock and he preferred a more generic name. But `InteractionMode` would be more descriptive and potentially reduce confusion. Thoughts @alexanderbock ?

Reply:
Yea, I think `InteractionMode` would be too restrictive, since Time for example is gated through this, but that's not really anything Interaction-like.    I could go with `Controller`?

Reply:
The motivation behind my comment is that the OpenSpaceEngine does a lot of different things, and so the 'mode' of the engine could apply to any of them. I'm thinking from the point of view of a developer who is studying the engine code and doesn't know much about camera paths/session recording. Another way to do this would be to keep the `Mode` name but have different names like `UserControl` rather than `Normal`.

Reply:
:+1: For `UserControl`

Reply:
👍